crates/rb-sys/src/stable_api/ruby_3_0.rs

```diff
@@ -96,6 +96,26 @@ impl StableApiDefinition for Definition {
         }
     }
 
+    #[inline]
+    unsafe fn rbasic_class(&self, obj: VALUE) -> VALUE {
+        let rbasic = obj as *const crate::RBasic;
+
+        (*rbasic).klass
+    }
+
+    #[inline]
+    unsafe fn rbasic_frozen_p(&self, obj: VALUE) -> bool {
+        // FIXME: Why this does not work?
```

Comment:
@ianks Please let me know if the `crate::rb_obj_frozen_p` version works to remove these comments.

Reply:
Let's re-implement them in the versioned APIs and do the bit-fiddling ourselves (to avoid dylib dispatch overhead). If that breaks things for truffle for now, that's OK. I want to re-think that definition once we see issues surface